packages/integration-tests/src/entities/BookReview.ts

```diff
@@ -1,9 +1,17 @@
-import { Reference } from "joist-orm";
-import { Author, BookReviewCodegen, bookReviewConfig } from "./entities";
+import { hasOneDerived, Reference } from "joist-orm";
+import { Author, BookReviewCodegen, bookReviewConfig, Publisher } from "./entities";
 
 export class BookReview extends BookReviewCodegen {
   // Currently this infers as Reference<BookReview, Author, undefined> --> it should be never...
   readonly author: Reference<BookReview, Author, never> = this.hasOneThrough((review) => review.book.author);
+
+  // This is kind of silly domain wise, but used as an example of hasOneDerived with a load hint. We don't
+  // technically have any conditional logic in `get` so could use a lens, but we want to test hasOneDerived.
+  readonly publisher: Reference<BookReview, Publisher, undefined> = hasOneDerived(
```

Comment:
can we unify how these are called? `hasOneThrough` is an instance method (still not a fan of this) and `hasOneDerived` is just an imported function.

Reply:
Right, hadn't gotten to making it an instance method...